### DISCUSSION TITLE
Component Re-render when other component state changes

### DISCUSSION THREAD

**Discussion by MuntasirTonmoy** [pmndrs/zustand#1862] [Q&A]
![Screenshot (77)](https://github.com/pmndrs/zustand/assets/96933032/2fb321b6-244e-4ee0-9fd1-ce1b07d4d4bb)
When I Click on the increment button the name list also re-render. I used shallow in the NameList component
CodeSandBox
https://codesandbox.io/s/zustand-gjksj8?file=/src/Counter.js

**Comment by dai-shi** [comment]
Looks like you already fixed it. 👍

**Reply by MuntasirTonmoy** [reply]
- Yeah what I figured out is that when I map, filter or find inside the store hook that means I directly subscribed to the store. 

- If any state changes in the store then every component that is directly subscribe to that particular store will re-render. 
- If I just pick single state then I only subscribe to that single state, so my component will re-render only when the single state changes.
